src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,401 @@
+                "pushkin@ukr.n",
+                ".145@gmail.com",
+                "pushkin@@mail.ru",
+                "1@111.11",
+                "pushkin.11",
+                "pushkin.1u",
+                "pushkin.u1",
+                "1@111.1u",
+                "1@111.1ru",
+                "1@111.ru1",
+                "1@ma~il.ru"));
+
+        List<String> specialCharsAndRusUkrLetters = getSpecialCharacters("\\");
+        specialCharsAndRusUkrLetters.removeAll(List.of("\\.", "\\_", "\\+", "\\-"));
+        specialCharsAndRusUkrLetters.addAll(getRussianAndUkrLetters());
+        specialCharsAndRusUkrLetters.forEach(
+                specialChar -> createCorrectEmails().forEach(
+                        email -> wrongEmails.add(email.replaceAll(
+                                REGROUPED_EMAIL_PATTERN, "$1".concat(specialChar).concat("$2$3$4$5$6"))))
+        );
+
+        return wrongEmails;
+    }
+
+    private static List<String> createCorrectPasswords() {
+        List<String> correctPasswords = new ArrayList<>();
+        String minCorrectPassword = "aA1";
+        correctPasswords.add(minCorrectPassword);
+
+        getSpecialCharacters("\\").forEach(
+                specialChar -> correctPasswords.add(minCorrectPassword.replaceAll(
+                        REGROUPED_PASSWORD_PATTERN, specialChar.concat("$1")))
+        );
+
+        return correctPasswords;
+    }
+
+    private static List<String> createWrongPasswords() {
+        String minCorrectPassword = "aA1";
+        List<String> wrongPasswords = new ArrayList<>(List.of(
+                "",
+                " ",
+                "w",
+                "1",
+                "W",
+                "`"
+        ));
+
+        List<String> tempList = new ArrayList<>();
+        IntStream.range(0, wrongPasswords.size()).forEach(i -> {
+            for (String wrongPassword : wrongPasswords) {
+                tempList.add(wrongPasswords.get(i).concat(wrongPassword));
+                tempList.add(wrongPasswords.get(i).concat(wrongPassword).concat(wrongPasswords.get(i)));
+            }
+        });
+
+        wrongPasswords.addAll(tempList);
+
+        wrongPasswords.addAll(List.of(
+                minCorrectPassword.concat(" "),
+                " ".concat(minCorrectPassword),
+                minCorrectPassword.concat(" A"),
+                "A ".concat(minCorrectPassword)
+        ));
+
+        getRussianAndUkrLetters().forEach(letter ->
+                createCorrectPasswords().forEach(password -> wrongPasswords.add(
+                        password.replaceAll(REGROUPED_PASSWORD_PATTERN, letter.concat("S1")))));
+
+        return wrongPasswords;
+    }
+
+    private static List<String> getCorrectPhoneNumbers() {
+        List<String> correctPhoneNumbers = new ArrayList<>();
+
+        String correctNumber = "+12-345-678-90-12";
+        String correctNumber1 = "13.345.678.90.13";
+
+        correctPhoneNumbers.add(correctNumber);
```

Comment:
Why do you need additional variables here and creation of list with 2 parameters? Make it simpler: return List.of()